pkg/kubectl/cmd/set/set_selector.go

```diff
@@ -213,6 +213,9 @@ func updateSelectorForObject(obj runtime.Object, selector metav1.LabelSelector)
 
 // getResourcesAndSelector retrieves resources and the selector expression from the given args (assuming selectors the last arg)
 func getResourcesAndSelector(args []string) (resources []string, selector *metav1.LabelSelector, err error) {
+	if len(args) == 0 {
```

Comment:
Sorry, not sure I understand.  Complete has the following line to parse the arguments which calls this function and causes the panic:

`o.resources, o.selector, err = getResourcesAndSelector(args)`

Are you saying this line is not needed?